### ISSUE TITLE
AssemblyScript Public Meeting #28 - January 13th, 2020

### ISSUE THREAD

**Issue by torch2424** [AssemblyScript/working-group#53]
# Date and Time

This public meeting will take place: **January 13th, 2020, 18:00 UTC (10:00 AM US PST, UTC -8)**

**Please Note**: We are skipping a week in December for the holidays :smile: 

# General Agenda

* Agenda Items from comments left on this Github issue
* Additional in-meeting comments / discussion
* [If time allows] Recap of [WebAssembly CG meeting](https://github.com/WebAssembly/meetings) if anyone attended

Feel free to comment on this issue if you have any agenda items you would like to bring up. 
Meeting Notes will be placed on this issue for those who cannot make the meeting.

# Meeting Information

We will use the AssemblyScript discord for meetings: https://discord.gg/vSBbJ5T . We will meet in the "meeting" voice channel :+1: 

Anyone who is contributing to the AssemblyScript project, building something with AssemblyScript, interested in the Assembly project or WebAssembly in general, is welcome to join! 😄

Notes will be taken by the host, and posted after the meeting. Notes are free to be edited through comments on the meeting notes at a later time.

**Comment by torch2424** [comment]
# Meeting Notes

* @torch2424 : Stuff I’m doing
   * Wasm Summit CFP, last week to do that
   * As-url library, hopefully out next week
   * Going to start AS JSON contributions today
   * Need to ask Fastly colks about Incremental GC
* @willemneal  : Working on some cool WASI recompilation stuff for Smart contracts
   * Wasm Summit talk may not make it by the 15th
   * Asbuild as put inside of asinit
      * @torch2424 : I think it makes sense :) People at Fastly like it as a default!
      * @dcodeIO : Wouldn’t we have to add as build to the compiler than?
      * @willemneal : Ideally we’d build on top of asinit. And, biasedly, we could then also add aspect
   * Could we move as-pect to the main org?
      * @dcodeIO : Yeah we could! But, I wouldn’t want to bias people to a specific testing solution. But I do agree it’s the best for AS right now
      * @willemneal : Ah true, but like Cargo test is a thing
   * Near Public thread on AS Support:
      * https://gov.near.org/t/planning-for-assemblyscript-support-and-integration-for-the-next-6-months-starting-jan21/345/11 
      * @torch2424 : All seemed positive on my end, outside looking in
* @torch2424 : Language Server
   * Mandy from Fastly wanted to help, will let you know once I have more concrete info Saul
   * @saulecabrera : Sounds good!
* @willemneal  & @MaxGraey : IR updates?
   * @MaxGraey : IR is kind of hard to tell without much usage, but making progress
* @MaxGraey : Worked on some BigNum support. Tests are passing
* @willemneal  & @jtenner : Moving as-pect into AS Org?
   * @jtenner : If we clean up the code, sure! :) 
* @jtenner : I need help with the Rtrace errors
   * @dcodeIO : Once the new runtime lands, this will no longer be an issue
   * @jtenner : Does the new runtime replace RTrace?
   * @dcodeIO : It’s similar, but it doesn’t do retain and releases anymore. 
* Sebastian: Any reason AS doesn’t have promises?
   * @willemneal : AS & Wasm isn’t Async
   * @jtenner : You can pass around promises using hostcalls, or asyncify in the binaryen pass
   * @torch2424 : You would need an event loop, which may come eventually
   * Sebastian: I’m working on a callback library, which isn’t quite working yet.
   * @jtenner & @MaxGraey : Yeah you could use some glue code from the browser, and simulate the promise async behavior from the browser for the simulated concurrency.
   * @MaxGraey : And this will require closures
* @jtenner : I’m going to pull out the as-pect Reflections into its own library
   * Any thoughts?
   * Sebastian: We could do something similar to the Typescript reflect metadata like used in the decorators?
* @jtenner : Can we have better stdlib functions for charcodes?
```
char("a");
18 as u16;
```
	   * Sebastian: I’m currently doing my own ease of use functions for char codes

```
type char = u16;
const CHAR_AT: char = u16("@".charCodeAt(0));
```
	  * @MaxGraey & @dcodeIO : It’s a little complicated, has some larger things to figure out

**Comment by torch2424** [comment]
Feel free to continue discussions from this meeting here :smile: 

Closing in favor of the next meeting: https://github.com/AssemblyScript/working-group/issues/54

**Comment by MaxGraey** [comment]
> Worked on some BigNum support. Tests are passing

Oh no))) I just mentioned `as-bignum` updates

**Comment by torch2424** [comment]
Oh oops! My apologies! Thank you for clarifying @MaxGraey ! :smile:

**Comment by MaxGraey** [comment]
I guess it's just my bad pronunciation 😂

**Comment by torch2424** [comment]
@MaxGraey Oh you were probably fine! I remember I was writing down notes from the thing before at the time, so I was like half listening :upside_down_face: :joy:
